File: solutions/javascript/blind-75/tree/sameTree.js
const isSameTree = (p, q) => {
    if (!p && !q) return true;
    // If either p or q is null, then we know there is a nonnull tree and a null tree, so we return false
    if (!p || !q) return false;
    
    const sameRootVal = p.val === q.val;
    
    const sameLeftSubtree = isSameTree(p.left, q.left);
    const sameRightSubtree = isSameTree(p.right, q.right);
    
    return sameRootVal && sameLeftSubtree && sameRightSubtree;
};

File: solutions/python/blind-75/tree/same_tree.py

class Solution:
    def isSameTree(self, p: Optional[TreeNode], q: Optional[TreeNode]) -> bool:
        if not p and not q:
            return True
        if not p or not q:
            return False
        
        same_root_val = p.val == q.val
        
        same_left_subtree = self.isSameTree(p.left, q.left)
        same_right_subtree = self.isSameTree(p.right, q.right)
        
        return same_root_val and same_left_subtree and same_right_subtree